### DISCUSSION TITLE
How to edit trashed (soft deleted) item

### DISCUSSION THREAD

**Discussion by msyadav88** [Laravel-Backpack/community-forum#866] [Q&A (Help)]
I want to edit a trashed item which is soft deleted, but it is giving 404 error. 
The url is like /admin/orders/7/edit 
Also the show page giving 404 /admin/orders/7/show

**Comment by jcastroa87** [comment]
Hello @msyadav88 

You can not edit a deleted item; to do this, you need to restore, by default, in a query is checked that is_deleted is null.

If you want to add this feature, you will need to override the query like this:

```
Model::withTrashed()->get();
```

Cheers.

**Reply by msyadav88** [reply]
I understand to not allow editing trashed item. But it is even not allow to show trashed item. The trashed item should allow atleast to view its detail.

**Reply by pxpm** [reply]
Hey @msyadav88 are you using Backpack TrashOperation ? https://backpackforlaravel.com/docs/6.x/crud-operation-trash#trash-a-single-item-pro

If you are please let me know how you configured so that I can try to reproduce your issue.

If not, make sure you have enabled soft deletes. You can do that in your `setupShowOperation` with: `CRUD::setOperationSetting('softDeletes', true)` and that your Model is using the SoftDeletes trait: `Illuminate\Database\Eloquent\SoftDeletes`

Let me know how it goes.

Cheers

**Reply by msyadav88** [reply]
Thank you Pedro for your help.
Yes I am using backpack TrashOperation, but just was missing "CRUD::setOperationSetting('softDeletes', true)". Now it working.
One thing now I am concern about is that, as we can't edit the trashed item, can we hide the edit icon for the trashed item?

**Reply by pxpm** [reply]
Hum, maybe I am missing your question, but you shouldn't have the edit button on the trashed items: 
![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/89948f70-8df1-4b82-8162-410b55d1f704)

You can have a look at it in our demo: https://demo.backpackforlaravel.com/admin/pet-shop/pet

Cheers

**Comment by msyadav88** [comment]
In the demo, I can not see even the show icon for trashed items.
In my case i can see the show as well as edit icon.

**Reply by pxpm** [reply]
Yes, I can see it too, but only for non-trashed items. 
The screenshot I sent earlier had the "trashed" filter applied. 
When the filter is not applied, I can just see the non-trashed items, and those can be edited, and you can preview them:
![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/7a06eba8-2044-4eec-bbf7-8fe3047d2f91)

Make sure you are not using global scopes in your model, or forcing some changes in the query in your controller. 

The code for the demo is here: https://github.com/Laravel-Backpack/demo/blob/36610772c8c90764e73b6c9fe91ec47184ebf447/app/Http/Controllers/Admin/PetShop/PetCrudController.php#L21

**Reply by msyadav88** [reply]
In my case, I can see preview as well as edit icon for trashed item. In screenshot order#7 is trashed.
![image](https://github.com/Laravel-Backpack/community-forum/assets/14232868/e82c3110-e0cf-405d-beab-e82c2903d295)

**Reply by pxpm** [reply]
Ohhh, I see, you disabled the Trash filter. 👌 thanks for the screenshot. 

Ok that's a different scenario. In that case you should prevent operations that shouldn't be allowed on trashed items, (like update, show etc).

I just noticed this was not clear at all in our docs, so I am not surprised by your confusion. Thanks for sticking with me on this one 🙏 

I've added that bit of information to our docs: https://github.com/Laravel-Backpack/docs/pull/557/files

When you disable the filter, you disable some setups the filter does, like preventing update/preview of trashed entries etc. In that case you should manually deny access the operations you use in your controller.

You should be fine with:
```php
public function setupTrashOperation()
{
    CRUD::setOperationSetting('withTrashFilter', false);
    CRUD::setAccessCondition(['update', 'show'], function($entry) {
        return ! $entry->trashed();
    });
}
```

Let me know if that helps.

Cheers

**Reply by msyadav88** [reply]
when I place CRUD::setAccessCondition it gives the error.
![image](https://github.com/Laravel-Backpack/community-forum/assets/14232868/eb953053-3e17-4e62-9d44-d4e626ff6cf3)

**Reply by pxpm** [reply]
Running `php artisan backpack:version` should give you a detailed overview of the installed package versions. 

I would suggest you update backpack/crud otherwise it's not possible to filter the table actions per entry, as `setAccessCondition` was the feature that enabled it. https://github.com/Laravel-Backpack/CRUD/releases/tag/6.2.4

Cheers 🙏

**Reply by msyadav88** [reply]
Ok I will sure consider to update version later. 
Right now when I try to use $entry = $this->crud->getCurrentEntry(); in setupShowOperation(), it gives error 404.
![image](https://github.com/Laravel-Backpack/community-forum/assets/14232868/0965f7cf-c41f-49d2-9dc3-958a64e212f1)


Nevermind , then I use request()->id, and its working.

**Comment by jcastroa87** [comment]
Great you was able to solve it.

I will close this issue now, but please feel free to re-open or create a new one if needed.

Cheers.
